File: fsharp/sum-of-multiples/SumOfMultiples.fs
module SumOfMultiples

let sum (numbers: int list) (upperBound: int): int =
   [1 .. upperBound - 1]
   |> List.filter (fun x ->
      List.map (fun n -> x % n = 0) numbers
      |> List.contains true
   )
   |> List.sum

File: haskell/sum-of-multiples/src/SumOfMultiples.hs
module SumOfMultiples (sumOfMultiples) where

import Data.Set (fromDistinctAscList, unions)

sumOfMultiples :: [Integer] -> Integer -> Integer
sumOfMultiples factors limit = sum $
  unions [fromDistinctAscList [0,f..limit-1] | f <- factors, f > 0]
